### PR TITLE
refactor(core): refactor core.module to typescript

### PR DESCRIPTION
**Pull request by christopherthielen**
- Remove uirouter.stateEvents.shim.js
- Refactor app initialization blocks to separate files

### CODE REVIEW

File: app/scripts/modules/core/src/bootstrap/bootstrap.module.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { module } from 'angular';
+
+import { OVERRIDE_REGISTRY } from 'core/overrideRegistry/override.registry';
+
+export const APPLICATION_BOOTSTRAP_MODULE = 'spinnaker.applicationBootstrap';
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/bootstrap/bootstrap.module.ts]: `spinnaker.core.applicationBootstrap`

File: app/scripts/modules/core/src/bootstrap/spinnaker.component.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  </div>
+  <div class="spinnaker-header navbar navbar-inverse">
+    <div ng-include="$ctrl.spinnakerHeaderTemplate"></div>
+  </div>
+
+  <div class="spinnaker-content">
+    <div ui-view="main" ng-if="!authenticating"></div>
+  </div>
+</div>
+<notifier></notifier>
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/bootstrap/spinnaker.component.ts]: Would be nice to indent the HTML here.

File: app/scripts/modules/core/src/bootstrap/stateChange.logger.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    $log.debug('$stateChangeStart', details);
+    const success = () => $log.debug('$stateChangeSuccess', details);
+    const failure = (error: any) => $log.debug('$stateChangeError', Object.assign(details, { error }));
+
+    transition.promise.then(success, failure);
+  });
+
+  $uiRouter.disposable({ dispose: () => subscription.unsubscribe() });
+});
+
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/bootstrap/stateChange.logger.ts]: extra newline

File: app/scripts/modules/core/src/routing/uirouter.stateEvents.shim.js [hunk 4]
```diff
@@ -1,146 +0,0 @@
-/**
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/routing/uirouter.stateEvents.shim.js]: There are a lot of `$stateChangeSuccess` and `$stateChangeStart` event subscriptions throughout the application. What fires those events now?

File: app/scripts/modules/core/src/core.module.ts [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+import 'jquery-ui';
+// Must come after jquery-ui - we want the bootstrap tooltip, JavaScript is fun
+import 'bootstrap/dist/js/bootstrap.js';
+
+import 'font-awesome/css/font-awesome.css';
+import 'react-select/dist/react-select.css';
+import 'ui-select/dist/select.css';
+
+import UI_ROUTER from '@uirouter/angularjs';
+require('@uirouter/angularjs/release/stateEvents');
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/core.module.ts]: @anotherchrisberry I deleted the copy of the shim from spinnaker and imported the one from the npm package here

File: app/scripts/modules/core/src/routing/uirouter.stateEvents.shim.js (hunk 4 above)
Comment (christopherthielen) [inline app/scripts/modules/core/src/routing/uirouter.stateEvents.shim.js]: https://github.com/spinnaker/deck/pull/3789/files#r120425866

File: app/scripts/modules/core/src/core.module.ts (hunk 5 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/core.module.ts]: ahhh, whew

**Pull request merged**